FunctionalTableDataTests/TableSectionChangeSetTests.swift

```diff
@@ -63,6 +63,31 @@ class TableSectionChangeSetTests: XCTestCase {
 		XCTAssertEqual(changes.deletedSections, IndexSet())
 	}
 
+	func testSectionNotReloadedWithEqualHeaders() {
+		let oldItems: [TableSection] = [TableSection(key: "section1", header: TestHeader(state: TestHeaderState(data: "state1")))]
+		let newItems: [TableSection] = [TableSection(key: "section1", header: TestHeader(state: TestHeaderState(data: "state1")))]
+		let changes = TableSectionChangeSet(old: oldItems, new: newItems, visibleIndexPaths: [])
+
+		XCTAssertTrue(changes.isEmpty)
+		XCTAssertEqual(changes.movedSections, [])
+		XCTAssertEqual(changes.insertedSections, IndexSet())
+		XCTAssertEqual(changes.reloadedSections, IndexSet())
+		XCTAssertEqual(changes.deletedSections, IndexSet())
+	}
```

Comment:
Can you also add rows to oldItems and newItems? So we can check the `else` when the headers are the same